src/Common/ISymbolExtensions.cs

```diff
@@ -215,30 +215,95 @@ internal static bool IsMoqRaisesMethod(this ISymbol symbol, MoqKnownSymbols know
+        return symbol.IsInstanceOf(knownSymbols.ICallbackRaises) ||
+               symbol.IsInstanceOf(knownSymbols.ICallback1Raises) ||
+               symbol.IsInstanceOf(knownSymbols.ICallback2Raises);
+    }
+
+    /// <summary>
+    /// Checks if the symbol is a Raises method from IReturns interfaces.
+    /// </summary>
+    /// <param name="symbol">The symbol to check.</param>
+    /// <param name="knownSymbols">The known symbols for type checking.</param>
+    /// <returns>True if the symbol is a returns Raises method; otherwise false.</returns>
+    private static bool IsReturnsRaisesMethod(ISymbol symbol, MoqKnownSymbols knownSymbols)
+    {
+        return symbol.IsInstanceOf(knownSymbols.IReturnsRaises) ||
+               symbol.IsInstanceOf(knownSymbols.IReturns1Raises) ||
+               symbol.IsInstanceOf(knownSymbols.IReturns2Raises);
+    }
+
+    /// <summary>
+    /// Checks if the symbol is a Raises method from IRaiseable interfaces.
+    /// </summary>
+    /// <param name="symbol">The symbol to check.</param>
+    /// <param name="knownSymbols">The known symbols for type checking.</param>
+    /// <returns>True if the symbol is a raiseable method; otherwise false.</returns>
+    private static bool IsRaiseableMethod(ISymbol symbol, MoqKnownSymbols knownSymbols)
+    {
+        return symbol.IsInstanceOf(knownSymbols.IRaiseableRaises) ||
+               symbol.IsInstanceOf(knownSymbols.IRaiseableAsyncRaisesAsync);
+    }
+
+    /// <summary>
+    /// Checks if a method is a Raises/RaisesAsync method by name and containing type.
+    /// This provides compatibility fallback when symbol-based detection fails.
+    /// </summary>
+    /// <param name="methodSymbol">The method symbol to check.</param>
+    /// <returns>True if the method is named Raises/RaisesAsync and is in a Moq.Language namespace; otherwise false.</returns>
+    private static bool IsRaisesMethodByName(IMethodSymbol methodSymbol)
+    {
         string methodName = methodSymbol.Name;
 
-        return (string.Equals(methodName, "Raises", StringComparison.Ordinal) ||
-                string.Equals(methodName, "RaisesAsync", StringComparison.Ordinal)) &&
-               containingTypeName?.Contains("Moq.Language", StringComparison.Ordinal) == true;
+        if (!IsRaisesMethodName(methodName))
+        {
+            return false;
+        }
+
+        string? containingTypeName = methodSymbol.ContainingType?.ToDisplayString();
+        return containingTypeName?.Contains("Moq.Language", StringComparison.Ordinal) == true;
+    }
+
+    /// <summary>
+    /// Checks if a method name matches the expected Raises method names.
+    /// </summary>
+    /// <param name="methodName">The method name to check.</param>
+    /// <returns>True if the method name is "Raises" or "RaisesAsync"; otherwise false.</returns>
+    private static bool IsRaisesMethodName(string methodName)
+    {
+        return string.Equals(methodName, "Raises", StringComparison.Ordinal) ||
```

Comment:
@coderabbitai Log an issue to correct this implementation. We should not be checking method names by strings, we should be using the pattern with `MoqKnownSymbols`